laserchicken/build_volume.py

```diff
@@ -0,0 +1,40 @@
+import inspect
+
+import laserchicken
+from laserchicken.volume_specification import Volume
+
+
+def create_volume_map():
+    """ Generate map of volume types that are available """
+    volume_map = {}
+    for name, obj in inspect.getmembers(laserchicken.volume_specification):
+        if inspect.isclass(obj) and issubclass(obj, Volume) and obj is not Volume:
+            volume_map[obj.TYPE] = obj
+    return volume_map
+
+
+VOLUMES = create_volume_map()
+
+
+def build_volume(vol_name, *args, **kwargs):
+    """
+    Return volume object from the volume name and the corresponding parameters
+
+    Example:
+    >>>> vol = build_volume('sphere', radius=5)
+
+    :param vol_name: name corresponding to the
+    :param args: optional non-keyword args to build the volume
+    :param kwargs: optional keyword args to build the volume
+    :return:
+    """
+    key = vol_name.lower()
+    _verify_volume_name(key)
+    vol = VOLUMES[vol_name.lower()]
+    return vol(*args, **kwargs)
```

Comment:
either create a new variable key and use it afterwards or don't declare the new variable at all. Also, if you choose to keep using the variable (and use it in line 33 as well), maybe it can have a more specific name although it is a bit of a tough one I must agree. Suggestions: 'volume_type_lower'

vol is not an accurate descriptive name as it contains a constructor for a specific volume class. As it is, it is probably unexpected that you suddenly start calling the vol variable with args and kwargs. More descriptive would be: 'volume_constructor'

Reply:
Ops, thanks, I have fixed this!